modules/optimization/test/tests/reporters/vector_math/tests

```diff
@@ -0,0 +1,28 @@
+[Tests]
+  issues = '#21885'
+  design = 'ParsedVectorRealReductionReporter.md ParsedVectorVectorRealReductionReporter.md ParsedVectorReporter.md ParsedScalarReporter.md'
+  [vectorMath]
+    type = CSVDiff
+    input = vectorMath.i
+    csvdiff = "vectorMath_out.csv vectorMath_out_vecvec_max_0001.csv vectorMath_out_vecvec_multiply_0001.csv
+               vectorMath_out_vecvec_sum_0001.csv vectorMath_out_vecvec_sqsum_0001.csv"
+    allow_test_objects = true
+    requirement = "The system shall be able to row sum a vector of vectors into a single vector, perform a dot product between two vectors and sum all of the entries of a single vector."
+  []
+  [errorRowSum]
+    type = RunException
+    input = vectorMath.i
+    cli_args = "Reporters/dataFromVofV/vector_of_vectors='101 201; 102 202; 103'"
+    allow_test_objects = true
+    expect_err = "Every vector in 'reporter_name=dataFromVofV/v_of_v' must be the same size."
+    requirement = "The system shall throw an error when the vectors being reduced are not the same size."
+  []
+  [errorDotProduct]
+    type = RunException
+    input = vectorMath.i
+    cli_args = "Reporters/vectorOperation/reporter_names='vecs/vec_a vecs/vec_b vec_d/vec_d'"
+    allow_test_objects = true
+    expect_err = "All vectors being operated on must be the same size."
+    requirement = "The system shall throw an error when the vectors being operated on are not the same size."
+  []
```

Comment:
These requirements need a little more context, something like:

- The system shall throw a reasonable error when reducing vectors and the vectors into a single vector and the input vectors are not the same size
- The system shall report a reasonable error when ...